QuickOpener/src/me/dsnet/quickopener/actions/Terminal.java

```diff
@@ -43,10 +45,23 @@ public void actionPerformed(ActionEvent e) {
                 customShellOpen(customShell.getValue(),path);
             }            
         } catch (Exception ex) {} 
-    }
+        } 
     
-    private void customShellOpen(String customShellPrefix, String path)throws Exception{
-        String fullCommand = customShellPrefix + path;
-        Runtime.getRuntime().exec(fullCommand);
+    private void customShellOpen(String customShell, String workingDir) throws Exception {
+        String shellTitle = "";
+        String fullCommand;
+        if (isWindows()) {
+            //support git bash this way
+            //cmd /c start /D workingDIR "" D:\tools\Git\bin\sh.exe --login -i
+            fullCommand = String.format("cmd /c start /D %s \"%s\" %s", workingDir, shellTitle, customShell);
```

Comment:
this would perhaps solve the problem of working directory with spaces in the name:
old:
fullCommand = String.format("cmd /c start /D %s \"%s\" %s", workingDir, shellTitle, customShell);
new:
fullCommand = String.format("cmd /c start /D \"%s\" \"%s\" %s", workingDir, shellTitle, customShell);